lib/tests/formats.nix

```diff
@@ -0,0 +1,186 @@
+# to run these tests:
+# nix-instantiate --eval --strict --read-write-mode nixpkgs/lib/tests/formats.nix
```

Comment:
yeah, `--read-write-mode` is a good indicator that this depends on import-from-derivation.

Reply:
Yeah, I'm in the process of making it not depend on that

Reply:
Done now